.github/workflows/build.yaml

```diff
@@ -45,7 +50,48 @@ on:
     branches: [ main ]
 
 jobs:
+  fetch_merge_commit_sha_from_lsp4ij_PR:
+    runs-on: ubuntu-latest
+    outputs:
+      pr_details: ${{ steps.extract.outputs.pr_details }}
+      checkout_name: ${{ steps.extract.outputs.checkout_name }}
+    env:
+      API_URL: https://api.github.com/repos/redhat-developer/lsp4ij/pulls
+      REF_LSP4IJ: ${{ inputs.refLsp4ij }}
+      LSP4IJ_BRANCH: ${{ inputs.lsp4ijBranch }}
+    name: Fetch Commit ${{ inputs.refLsp4ij || '' }}
+    steps:
+      - name: Extract Merge Commit SHA
```

Comment:
We already have code in cronJob.yaml to extract the merge commit SHA. Is there any way that we can re-use that code somehow instead of re-doing it here?

Reply:
@TrevCraw , I tried investigating ways to reuse code from the `cronJob.yaml` file. We only require a specific portion of the code for the `fetch_all_pull_request_shas` job, and we do not need the complete job or the other three jobs. During my investigation, I found the following:

GitHub Actions does not support reusing a single job or some portion of a job from one workflow in another. Instead, we can achieve this by creating a new reusable workflow file that includes only the specific job(s) we want to reuse, but we can't create a common code for both the jobs that is, we use different values are being passed in the jobs, such as `checkout_name` in the `fetch_merge_commit_sha_from_lsp4ij_PR` job (from `build.yaml`) and `is_empty` in the `fetch_all_pull_request_shas` job. The `pr_details` variable is common to both jobs but has different values in each case.